### DISCUSSION TITLE
Prelude.Aff vs Prelude.AffMaybe

### DISCUSSION THREAD

**Discussion by Malgefor** [louthy/language-ext#1237] [Q&A] [answered]
Hi all,
When lifting a Task/Task<T> into Aff-monad, I see both `Prelude.Aff` and `Prelude.AffMaybe` methods who can perform that operation.
Is there a difference / best-practice?

Thanks in advance!

**Comment by louthy** [comment, marked as answer]
`AffMaybe(...)` allows you to return a `Fin<A>`, that gives you the opportunity to lift a lambda that could fail (i.e. return `Fin.Succ` or `Fin.Fail`).  `Aff(...)` doesn't allow that, so it expects the lifted function to always succeed (i.e. it's pure). 

Obviously, you _could_ throw an exception in the `Aff` lifted function, so it's not entirely clear cut.   However, best practice is to avoid throwing exceptions for 'known' errors (i..e. exceptions are only for exceptional events, like running out of memory).

**Reply by Malgefor** [reply]
I see, thanks for the reply! :)
